### DISCUSSION TITLE
Fails to start in vscode

### DISCUSSION THREAD

**Discussion by nikita-fuchs** [erlang-ls/erlang_ls#1363] [Support Requests]
**Describe the bug**
In around 90% of the cases, the erlang LS fails to start in VS code for me. On the last sufccessful start, VScode shows a activation time of a little over 20 seconds.

**To Reproduce**
Start VScode with erlang LS installed.

**Expected behavior**
the LS would start normally.

**Actual behavior**
VScode shows an error notification that the erlang LS could not be started. I am not sure how to gather more information on this sparse error message unfortunately.

**Context**
 - `erlang_ls` version (tag/sha): v0.0.35 (vscode plugin version info)
 - Editor used: vscode 1.69
 - LSP client used: couldn't find any info on that, would be happy to add this if someone gives me a hint.

**Comment by robertoaloi** [comment]
Hi @nikita-fuchs and sorry for the late reply. Did you try to run the Erlang LS escript which is shipped with the extension manually? There could be a problem with your Erlang installation / version. What version of OTP are you running?

Also, you can check the [troubleshooting](https://erlang-ls.github.io/troubleshooting/) section of the docs for where to find the Erlang LS logs. In your case, though, it looks like the language server does not even start.

Another place you could look at is the Output panel in VS Code (next to the terminal), where you may find some hints on why the server cannot start correctly.

**Comment by robertoaloi** [comment]
Moving this from an "issue" to a "discussion" since it seems related to a specific Erlang installation. If we find some actual issue in Erlang LS (or the docs!) we can open an actual issue.

**Comment by nikita-fuchs** [comment]
Thanks for getting back to me ! :) I'm afraid I have to let this one rest a bit if you don't mind, because everything works fine now, magically. The next time this error appears, I will check `Output` panel again. I remember that one saying something, but I couldn't catch it before it was gone.

**Comment by nikita-fuchs** [comment]
It happened again: 
<img width="1018" alt="image" src="https://user-images.githubusercontent.com/11761828/188131854-b8f8680f-bb07-4c54-aec3-281089069454.png">

Is there anything I can do to debug this any further? I'm on `Erlang/OTP 24 [erts-12.2.1]`

**Comment by jcsawyer123** [comment]
Just wanted to follow up on this instead of making a duplicate. Having the exact same issue,  the only error presented in logs is the `launching server using command escript failed`. 

Have also tried disabling all other plugins.

MacOS Monterey v12.2.1
VsCode 1.71.2
Erlang/OTP 21 [erts-10.3.5.16]

**Comment by nikita-fuchs** [comment]
Today it started happening again, I can't figure out what I'm doing differently: 
<img width="473" alt="image" src="https://user-images.githubusercontent.com/11761828/196396857-f5a51e75-c760-4cc3-b647-00da51c974a5.png">

The output is just the same again: 
<img width="351" alt="image" src="https://user-images.githubusercontent.com/11761828/196396956-048a13c7-9c3c-4d20-9209-12fab8b0135a.png">

MacOS Monterey 12.6, 

Version: 1.72.2
Commit: d045a5eda657f4d7b676dedbfa7aab8207f8a075
Date: 2022-10-12T22:16:30.254Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 21.6.0
Sandboxed: No

**Comment by AnufrievDE** [comment]
Faced the same issue, on windows 10, with wsl2.

I've added the debug print to erlang_ls vscode extension - ~/.vscode-server/extensions/erlang-ls.erlang-ls-0.0.39/client/out/client.js:

```
        vscode_1.window.showErrorMessage('Res: ' + JSON.stringify(res));
        vscode_1.window.showErrorMessage('Could not start Language Server. Error: ' + res.stdout); 
 ```
 
I've found that the error returned in res variable is "enoent", which means that there is no path to 'escript' (installed within erlang with asdf in wsl2).

When looking for "enoent vscode issues", found that such an issue happend for different VS Code extentions, e.g.: https://stackoverflow.com/questions/69166487/process-failed-spawn-c-windows-system32-cmd-exe-enoent-jest-process-watch-tes.

Advice from this topic helped me - erlang_ls launched well when I open VS Code directory from wsl terminal with - ```code ./example_erl_project &```, nor from taskbar or explorer.

**Comment by cleanmountain** [comment]
Had the same issue on Ubuntu, but found that I had just forgotten to activate my `kerl` OTP installation - i.e. I had no Erlang installation in my PATH. So I just added this to my `.zshrc`

```
source ~/.kerl/installations/26.2.5.13/activate
```

...and it worked after restarting vscode.
